### PR TITLE
fix: pass file id for direct editing and fail y.js provider setup if none was passed

### PR DESCRIPTION
**Pull request by juliushaertl**
### 📝 Summary

When setting up the WebsocketProvider we still used to create them even if no file id was passed. This is actually the case for direct editing trough android/iOS and would mean that if some browser states in the webview is still present it could mix up documents.

This is basically a follow up to https://github.com/nextcloud/text/commit/b2186da8610a2762f4b155d1452402eac3423cdd

* Possibly fixes https://github.com/nextcloud/text/issues/4112
* Possibly fixes parts of https://github.com/nextcloud/text/issues/4100

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

**Comment by juliushaertl** [comment]
Note that the error behaviour is currently quite bad as when this happens the Editor.vue is just empty, will see if we can improve that as well.

File: src/services/SyncServiceProvider.js [hunk 1]
```diff
@@ -33,6 +33,9 @@ import { logger } from '../helpers/logger.js'
  * @param {object} options.initialSession - initialSession to start from
  */
 export default function createSyncServiceProvider({ ydoc, syncService, fileId, initialSession }) {
+	if (!fileId) {
```
Comment (mejo-) [inline src/services/SyncServiceProvider.js:36]: Maybe a short comment would be helpful here for future reference?

**Review by mejo- — COMMENTED** [review]
Just a minor comment.

And to be honest I don't fully understand the situation. So directediting sessions didn't pass a `fileId` before the change to `DirectEditing.vue` from this PR, which lead to `WebsocketProvider` instances with similar identifiers, which in turn could result in content from leftover sessions getting mixed into the new session. Is this roughly the scenario that this PR aims to fix?

**Comment by mejo-** [comment]
/backport 23a46cd582cca8224d0fa6beca60b9e9c76f80c5 to stable26

**Comment by max-nextcloud** [comment]
Great catch @juliushaertl and thanks for the quick review and backport @mejo- ! :heart:

**Pull request merged**